src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -41,7 +41,16 @@ enum class LQPNodeType {
 
 enum class LQPChildSide { Left, Right };
 
+// Describes the parent of a Node and which of the parent's children a node is
+struct LQPParentRelation {
+  std::shared_ptr<AbstractLQPNode> parent;
```

Comment:
A reference as in `std::shared_ptr<AbstractLQPNode>& parent` or `AbstractLQPNode& parent`? Since when are we storing references to things as references and not shared_ptrs?

Reply:
My idea was that the LQPParentRelation should not have any influence on the lifetime of the nodes. Is it true that we don't store references anywhere else?

Reply:
I think we don't.